### PR TITLE
[docs] Organize migration pages

### PR DESCRIPTION
**Pull request by flaviendelangle**
[Doc preview](https://deploy-preview-6480--material-ui-x.netlify.app/x/migration/migration-data-grid-v5/)

## Follow up

- Add redirections for the existing pages on the core

### CODE REVIEW

File: docs/data/pages.ts [hunk 1]
```diff
@@ -236,6 +233,34 @@ const pages: MuiPage[] = [
...
+  {
+    pathname: '/x/migration',
+    title: 'Migration',
+    icon: 'BookIcon',
+    children: [
+      {
+        pathname: '/x/migration',
+        subheader: 'Upgrade to v6',
+        children: [
+          { pathname: '/x/migration/migration-data-grid-v5', title: 'Breaking changes: Data Grid' },
```
Comment (flaviendelangle) [inline docs/data/pages.ts:245]: #6472 would come here

File: docs/data/pages.ts [hunk 2]
```diff
@@ -236,6 +233,34 @@ const pages: MuiPage[] = [
...
+        ],
+      },
+      {
+        pathname: '/x/migration',
+        subheader: 'Earlier versions',
+        children: [
+          {
+            pathname: '/x/migration/migration-pickers-lab',
+            title: 'Migration from lab to v5 (pickers)',
```
Comment (flaviendelangle) [inline docs/data/pages.ts:254]: The correct naming would be "date & time pickers" but that's very long for the nav bar

File: docs/data/pages.ts (hunk 1 above)
Comment (alexfauquette) [reply docs/data/pages.ts:280]: Should be added to `docs/public/_redirects`

```
/:lang/x/react-data-grid/migration-v4/ /:lang/x/migration/migration-data-grid-v4/ 301
/:lang/x/react-data-grid/migration-v5/ /:lang/x/migration/migration-data-grid-v5/ 301
/:lang/x/react-date-pickers/migration-lab/ /:lang/x/migration/migration-pickers-lab/ 301
```

**Review by alexfauquette — COMMENTED** [review]
Nice to have a dedicated section for them

**Pull request merged**